src/ChangelogEmbed/Changelog_Parser.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Changelog Parser.
+ * 
+ * TODO: Test it with multiple log files from different repositories.
+ *
+ * @since 2.0.0
+ *
+ * @package StellarWP\ChangelogEmbed
+ */
+
+namespace StellarWP\ChangelogEmbed;
+
+/**
+ * Handles parsing of changelog files in WordPress plugin format.
+ *
+ * @since 2.0.0
+ */
+class Changelog_Parser {
+	/**
+	 * Parse changelog content.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $content      Raw changelog content.
+	 * @param int    $max_versions Maximum number of versions to return. Default is 5.
+	 *
+	 * @return array Parsed changelog data.
+	 */
+	public function parse( string $content, int $max_versions = 5 ): array {
+		// Initialize variables.
+		$changelog_data = [];
+		$current_version = null;
+
+		// Split content into lines.
+		$lines = explode( "\n", $content );
+
+		foreach ( $lines as $line ) {
+			$line = trim( $line );
+
+			// Skip empty lines.
+			if ( empty( $line ) ) {
+				continue;
+			}
+
+			// Check for version headers (e.g., "= [4.21.1] =").
+			if ( preg_match( '/^=\s*\[([^\]]+)\]\s*=(?:\s*-\s*(.+))?$/i', $line, $matches ) ) {
```

Comment:
We'll need to ensure this also accounts for the new format that LD Core is using. 

Example: `= [4.23.2.1] 2025-07-24 =`

A quick test shows that it doesn't. 🫠